### PR TITLE
[SPARK-24694][K8S] Pass all app args to integration tests

### PR DESCRIPTION
**Pull request by skonto**
## What changes were proposed in this pull request?
- Allows to pass more than one app args to tests.
## How was this patch tested?
Manually tested it with a spark test that requires more than on app args.

### CODE REVIEW

**Comment by skonto** [comment]
@ssuchter @liyinan926 psl review. This is trivial, unless I am missing something.

**Comment by ssuchter** [comment]
So this changes behavior, I think. In the old behavior, if the args were ['a', 'b'] then you'd get a single arg of ['a b'] passed through, and with this you'd get ['a', 'b'].

This new behavior seems better, I'm just trying a bit to remember why we had the old behavior.

**Comment by ssuchter** [comment]
I see why the old behavior was there. I made a minimal change to some existing code to fix a bug:

https://github.com/ssuchter/spark/commit/1d8a265d13b65dcec8db11a5be09d4a029037d2c

but this new way is better.

Question: In this new way, do we even need the test for appArguments.appArgs.length > 0? Could we just use appArguments.appArgs?

**Comment by ssuchter** [comment]
BTW, for committers - I think this patch is good to merge.

**Comment by skonto** [comment]
@ssuchter yes if you check the jira, the old behavior does not work with more than one args. In the future might be a problem.

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesTestComponents.scala [hunk 1]
```diff
@@ -106,7 +106,7 @@ private[spark] object SparkAppLauncher extends Logging {
     val sparkSubmitExecutable = sparkHomeDir.resolve(Paths.get("bin", "spark-submit"))
     logInfo(s"Launching a spark app with arguments $appArguments and conf $appConf")
     val appArgsArray =
-      if (appArguments.appArgs.length > 0) Array(appArguments.appArgs.mkString(" "))
```
Comment (vanzin) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesTestComponents.scala]: Doesn't this make `appArgsArray` basically redundant?

Comment (skonto) [reply resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesTestComponents.scala]: Yes unless you pass a null... I updated my comment above ^^.

**Review by felixcheung — COMMENTED** [review]
looks reasonable. is there a way to add test for this?

**Comment by skonto** [comment]
@felixcheung we could call a test with more than one parameters like: https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/MultiBroadcastTest.scala

**Comment by skonto** [comment]
@felixcheung I think its ok to merge. In the future when we will add tests passing more params we can verify it easily, but it is better than before.

**Comment by skonto** [comment]
@foxish @srowen  gentle ping.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**